ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -352,6 +353,74 @@ PrjFS_Result PrjFS_WritePlaceholderFile(
     return PrjFS_Result_EIOError;
 }
 
+PrjFS_Result PrjFS_UpdatePlaceholderFileIfNeeded(
+    _In_    const char*                             relativePath,
+    _In_    unsigned char                           providerId[PrjFS_PlaceholderIdLength],
+    _In_    unsigned char                           contentId[PrjFS_PlaceholderIdLength],
+    _In_    unsigned long                           fileSize,
+    _In_    uint16_t                                fileMode,
+    _In_    PrjFS_UpdateType                        updateFlags,
+    _Out_   PrjFS_UpdateFailureCause*               failureCause)
+{
+#ifdef DEBUG
+    std::cout
+        << "PrjFS_UpdatePlaceholderFileIfNeeded("
+        << relativePath << ", "
+        << (int)providerId[0] << ", "
+        << (int)contentId[0] << ", "
+        << fileSize << ", "
+        << std::oct << fileMode << std::dec << ", "
+        << std::hex << updateFlags << std::dec << ")" << std::endl;
+#endif
+    
+    PrjFS_Result result = PrjFS_DeleteFile(relativePath, updateFlags, failureCause);
```

Comment:
Is this just a temporary approach? Will we be able to update the existing placeholder in place?

Reply:
I was not thinking this (delete+add) would be a temporary approach:

- ProjFS on Windows internally does a delete+add when updating placeholders
- Git does a delete+add when updating the working tree as part of a checkout

We do need to compare content ID, but I was thinking we'd still with delete+add so that applications see the same events with VFSForGit as they do with Git.